Shopify/cli
# Detect Global CLI: update prompts and show warning if needed

### WHY are these changes introduced?
There are prompts that tell you to run specific commands, this prompts should be accurate when using a Global CLI
If using a Global CLI in a project that also has a local dependency, we want to show a warning.

Fixes https://github.com/Shopify/develop-app-management/issues/1657 and https://github.com/Shopify/develop-app-management/issues/1658



### WHAT is this pull request doing?
- Add a way to detect if the current process is being run globally
- If Global: any prompt that includes a command reference should be displayed in the global format
- If Global and running in an app project that also has the CLI as a local dependency: show a warning about it.
<img width="1098" alt="Captura de pantalla 2024-03-14 a las 10 13 01" src="https://github.com/Shopify/cli/assets/3757193/efffcdad-8ffb-4cbf-bc64-736fb89a5ba2">





### How to test your changes?
- Install the global CLI with `npm i -g @shopify/cli@0.0.0-experimental-20240314124741`
- Run `shopify app build` (or any app command) in an existing app folder -> you should see the warning
- Run `shopify app config use --reset` in that folder, you should see a message with a command, the command should be formatted for the global CLI.

You can do the same tests for non-global CLI and it should continue working as it does now.



### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
